[W5.11][T13-1] Adrian Tan

QzSG opened the pull request:
[[LO-2KLoC]](https://github.com/nusCS2113-AY1819S1/addressbook-level2/blob/master/docs/LearningOutcomes.adoc#work-in-a-2kloc-code-base-lo-2kloc) Add a feature to AddressBook
- Adds `sort` command to Addressbook-2 along with relevant unit tests to fulfill `W5.11` 🏆

QzSG committed:
:sparkles: Add new Sort Command

Adds Sort Command to Help and Command Parser.

QzSG committed:
Add Simple Unit Test for Sort Command

Add Simple Unit Test to check if sort command correctly sorts an unsorted address book

Refactor sorting logic out of execute method for Sort Command

Note: Currently Implementation of Sort is more like a virtual view. Underlying data is not modified

TODO: Make sort command sort actual  data file.

QzSG committed:
:sparkles: Enhance Sort Command :recycle: :

Make Sort Command inherit from List Command, thereby also simplifying Unit Test :smile:

Make Sort Command persistent

QzSG committed:
:memo: Update User Guide :pencil2:

Adds Sorting command description and usage to UserGuide.adoc

ewaldhew commented on src/seedu/addressbook/parser/Parser.java:
```diff
@@ -11,17 +11,7 @@
-import seedu.addressbook.commands.ListCommand;
-import seedu.addressbook.commands.ViewAllCommand;
-import seedu.addressbook.commands.ViewCommand;
+import seedu.addressbook.commands.*;
```
`import` statements should be explicit. Configure your IntelliJ to not collapse imports like this.
`Preferences` > `Editor` > `Code Style` > `Java` > `Imports tab set Class count to use import with '*'`

ewaldhew commented on test/java/seedu/addressbook/commands/SortCommandTest.java:
```diff
@@ -0,0 +1,63 @@
+        sortCommand.setData(unsortedAddressBook, listWithUnsortedPersons);
+
+        // sortedAddressBook is already sorted when created.
+        SortCommand expectedCommand = new SortCommand();
```
This is never used..

QzSG replied:
Looks like I left artifacts when renaming stuff, thanks for the heads up...

ewaldhew reviewed (commented):
Looks fine to me, but do remember to configure coding style in IntelliJ. Close this PR after reading.

QzSG committed:
Fixes based on Code Review

Parser.java : Made imports explicit

SortCommandTest.java : Clean up unused artifacts. Fix naming

QzSG commented:
Closing PR

luhan02 commented:
Great enhancement.

ewaldhew commented on src/seedu/addressbook/commands/SortCommand.java:
```diff
@@ -0,0 +1,32 @@
+            + "Example: " + COMMAND_WORD;
+
+
+
```
Use blank lines sparingly. It's good to have blank lines to separate logical code blocks, but too many blank lines will increase the height of the code, which increases the amount of scrolling.

ewaldhew commented on docs/UserGuide.adoc:
```diff
@@ -72,6 +72,12 @@ Examples:
+Sorts all persons and persists to storage.
+Also shows a list of all persons, along with their non-private details, in the address book after sorting. +
+Format: `sort`
+
```
Good job updating the user guide

ewaldhew reviewed (commented):
Great work, and with updating the JUnit tests as well. 👍 Close PR after reading

QzSG commented:
Closing PR
